src/arkreactor/VM/VM.cpp

```diff
@@ -138,8 +138,11 @@ namespace Ark
         {
             // put it in the global frame, aka the first one
             auto it = std::find(m_state->m_symbols.begin(), m_state->m_symbols.end(), std::string(map[i].name));
-            if (it != m_state->m_symbols.end())
-                (*m_locals[0]).push_back(static_cast<uint16_t>(std::distance(m_state->m_symbols.begin(), it)), Value(map[i].value));
+            if (it == m_state->m_symbols.end())
+            {
+                it = m_state->m_symbols.insert(m_state->m_symbols.end(), map[i].name);
+            }
+            m_locals[0]->push_back(static_cast<uint16_t>(std::distance(m_state->m_symbols.begin(), it)), Value(map[i].value));
```

Comment:
This isn't a problem, because if the symbol from the plugin isn't in the symbol table, it means that it isn't used, thus we shouldn't clutter said table with those values.

The issue 337 is about loading a release plugin with a debug build of Ark

Reply:
Let me have another look.

Is there an automated set of unit tests I can run for this project?
So that I can make sure my changes don't break anything.

Reply:
Also Note:

There is no expectation of the language that compilation units built with different flags will work together at all.
If you have release build objects and debug build objects, they should not be linked or laded together into the same executable (flags can do very strange things to layout).

We should probably consider a naming convention that differentiates Debug files and Release files (i.e Add "D" onto the end of a module that is build with Debug flags).

Reply:
Yes, there is a test suite ran by the CI, in tests/
It needs the executable to be built, and the console module to be available as well as the standard library.

Reply:
As for the compiler/flags problem, that's the root of the plugin loading errors, though we haven't come to any decision/solution to fix it, yet (hence the issue).

The problem is that we want to be able

Reply:
OK I just did a bunch of refactoring on the `State` class to understand how it is used.
There are only 2 places were things get added to the "Global Scope" both have the same test. This effectively means that nothing is ever added to the "Global Scope" what am I missing?

Reply:
> Yes, there is a test suite ran by the CI, in tests/ It needs the executable to be built, and the console module to be available as well as the standard library.

How do I run those manually from the command line?

Reply:
> As for the compiler/flags problem, that's the root of the plugin loading errors, though we haven't come to any decision/solution to fix it, yet (hence the issue).

OK.
So the current module are `*.arkm` and `*.ark` (not sure what the difference is yet).

What is the processes we use for getting a new standard.
I don't want to work on a solution like this unless there is agreement on the direction.

I would suggest (but easy to go a different direction).
`* [_D].ark[m]`

_D: Is the debug version. Only loaded by the debug version of Arc.

Reply:
> How do I run those manually from the command line?

You should be able to run them with `build/arkscript -L ./lib tests/arkscript/unittests.ark`

As for the issue forget it I misread the issue number and assumed it (337) was 259. The og fix is valid though it's modifying the State which is a critical ressource (can be/is shared between multiple VM)

> What is the processes we use for getting a new standard.

We go through RFC on ArkScript-lang/rfc so that we have a clear view and technical understanding of a feature like this one (many features & bug fixes don't need RFC luckily)